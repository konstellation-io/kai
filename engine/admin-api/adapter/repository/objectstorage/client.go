package objectstorage

import (
	"fmt"

	"github.com/konstellation-io/kai/engine/admin-api/adapter/config"
	"github.com/minio/madmin-go/v3"
	"github.com/minio/minio-go/v7"
	"github.com/minio/minio-go/v7/pkg/credentials"
	"github.com/spf13/viper"
)

func NewMinioClient() (*minio.Client, error) {
	var (
		minioKeyID     = viper.GetString(config.MinioRootUserKey)
		minioKeySecret = viper.GetString(config.MinioRootPasswordKey)
	)

	client, err := minio.New(viper.GetString(config.MinioEndpointKey), &minio.Options{
		Creds:        credentials.NewStaticV4(minioKeyID, minioKeySecret, ""),
		BucketLookup: minio.BucketLookupPath,
	})
	if err != nil {
		return nil, fmt.Errorf("initializing MinIO client: %w", err)
	}

	return client, nil
}

func NewAdminMinioClient() (*madmin.AdminClient, error) {
	var (
		minioKeyID     = viper.GetString(config.MinioRootUserKey)
		minioKeySecret = viper.GetString(config.MinioRootPasswordKey)
	)

	client, err := madmin.New(viper.GetString(config.MinioEndpointKey), minioKeyID, minioKeySecret, false)
	if err != nil {
		return nil, fmt.Errorf("initializing MinIO admin client: %w", err)
	}

	return client, nil
}
