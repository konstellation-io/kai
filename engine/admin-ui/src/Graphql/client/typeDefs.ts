import { AccessLevel } from 'Graphql/types/globalTypes';
import { NotificationType } from 'Graphql/client/models/Notification';

export interface AddNotificationInput {
  id: string;
  message: string;
  type: NotificationType;
  timeout: number;
  to: string;
  typeLabel?: string;
}

export interface AddLogTabInput {
  runtimeId: string;
  runtimeName: string;
  versionId: string;
  versionName: string;
  nodes: NodeSelection[];
}

export interface NodeSelection {
  workflowName: string;
  nodeNames: string[];
  __typename: string;
}

export interface LogPanelFilters {
  dateOption?: string;
  startDate?: string;
  endDate?: string;
  nodes?: NodeSelection[];
  search?: string;
  levels?: string[] | null;
  __typename: 'logTabFilters';
}

export interface LogPanel {
  runtimeId: string;
  runtimeName: string;
  versionId: string;
  versionName: string;
  uniqueId?: string;
  filters?: LogPanelFilters;
}

export interface UserSettingsFilters {
  email: string | null;
  accessLevel: AccessLevel | null;
  __typename: 'UserSettingsFilters';
}

export interface RemoveNotificationInput {
  id: string;
}
