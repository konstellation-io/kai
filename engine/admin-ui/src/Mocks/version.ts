import { NodeStatus, VersionStatus } from 'Graphql/types/globalTypes';

import GetConfigurationVariablesQuery from 'Graphql/queries/getConfigurationVariables';
import GetVersionWorkflowsQuery from 'Graphql/queries/getVersionWorkflows';
import GetVersionNodeStatusQuery from 'Graphql/subscriptions/watchVersionNodeStatus';
import NodeStatusSubscription from 'Graphql/subscriptions/watchVersionNodeStatus';
import PublishVersionMutation from 'Graphql/mutations/publishVersion';
import StartVersionMutation from 'Graphql/mutations/startVersion';
import StopVersionMutation from 'Graphql/mutations/stopVersion';
import UnpublishVersionMutation from 'Graphql/mutations/unpublishVersion';

export const runtime = {
  id: 'runtimeId',
  name: 'runtimeName',
  description: 'Some Description',
  measurementsUrl: 'measurementsUrl',
  __typename: 'Runtime'
};
export const version = {
  __typename: 'Version',
  id: 'versionId',
  name: 'versionName',
  description: 'versionDescription',
  status: VersionStatus.STARTED,
  creationDate: '2020/01/01',
  creationAuthor: {
    __typename: 'User',
    id: 'userId',
    email: 'user@domain.com'
  },
  publicationDate: '2020/01/01',
  publicationAuthor: {
    __typename: 'User',
    id: 'userId',
    email: 'user@domain.com'
  },
  config: {
    vars: [],
    completed: false
  },
  hasDoc: false,
  errors: []
};
export const workflow = {
  __typename: 'Workflow',
  id: 'workflowId',
  name: 'Workflow Nane',
  nodes: [
    {
      __typename: 'Node',
      id: 'nodeId',
      name: 'Node Name',
      status: NodeStatus.STARTED
    }
  ],
  edges: [
    {
      __typename: 'Edge',
      id: 'edgeId',
      fromNode: 'node1',
      toNode: 'node2'
    }
  ]
};

export const confVarsMock = {
  request: {
    query: GetConfigurationVariablesQuery,
    variables: {
      versionId: 'id'
    }
  },
  result: {
    data: {
      version: {
        status: 'STARTED',
        config: {
          completed: true,
          vars: [
            {
              key: 'var2',
              value: 'value1',
              type: 'VARIABLE'
            },
            {
              key: 'var1',
              value: 'value2',
              type: 'VARIABLE'
            },
            {
              key: 'var3',
              value: 'value3',
              type: 'FILE'
            }
          ]
        }
      }
    }
  }
};

export const workflowsMock = {
  request: {
    query: GetVersionWorkflowsQuery
  },
  result: {
    data: {
      version: {
        name: 'version01',
        status: 'STARTED',
        configurationCompleted: true,
        workflows: [
          {
            name: 'workflow01',
            nodes: [
              {
                id: 'node01',
                name: 'one node',
                status: 'STARTED'
              },
              {
                id: 'node02',
                name: 'another node',
                status: 'STARTED'
              }
            ],
            edges: [
              {
                id: 'edge01',
                fromNode: 'node01',
                toNode: 'node02'
              }
            ]
          }
        ]
      }
    }
  }
};

export const errorMorkflowsMock = {
  request: {
    query: GetVersionWorkflowsQuery
  },
  error: 'Some error',
  result: {
    data: null
  }
};

export const nodeStatus = {
  request: {
    query: GetVersionNodeStatusQuery
  },
  result: {
    data: {
      versionNodeStatus: [
        {
          date: '2020-01-01',
          nodeId: 'nodeId1',
          status: 'STARTED',
          message: 'some message 1'
        },
        {
          date: '2020-01-01',
          nodeId: 'nodeId2',
          status: 'STARTED',
          message: 'some message 2'
        }
      ]
    }
  }
};

export const nodeStatusMock = {
  request: {
    query: NodeStatusSubscription
  },
  result: {
    data: {
      versionNodeStatus: {
        date: '2020/01/01',
        nodeId: 'node02',
        status: 'STARTED',
        message: 'some message'
      }
    }
  }
};

export const publishVersionMock = {
  request: {
    query: PublishVersionMutation
  },
  result: {
    data: {
      version
    }
  }
};

export const startVersionMock = {
  request: {
    query: StartVersionMutation
  },
  result: {
    data: {
      version
    }
  }
};

export const stopVersionMock = {
  request: {
    query: StopVersionMutation
  },
  result: {
    data: {
      version
    }
  }
};

export const unpublishVersionMock = {
  request: {
    query: UnpublishVersionMutation
  },
  result: {
    data: {
      version
    }
  }
};
