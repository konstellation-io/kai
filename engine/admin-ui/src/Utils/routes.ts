export const buildRoute = (route: string, versionId: string = ''): string =>
  route.replace(':versionId', versionId);
