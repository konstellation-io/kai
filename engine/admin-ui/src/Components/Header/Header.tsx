import React, { FunctionComponent, ReactElement, memo } from 'react';

import { GetMe } from 'Graphql/queries/types/GetMe';
import Settings from 'Components/Settings/Settings';
import { get } from 'lodash';
import { loader } from 'graphql.macro';
import styles from './Header.module.scss';
import { useQuery } from '@apollo/client';
import ROUTE from "Constants/routes";
import { Link, useRouteMatch } from 'react-router-dom';
import cx from 'classnames';

const GetMeQuery = loader('Graphql/queries/getMe.graphql');

type Props = {
  children?: ReactElement | ReactElement[] | null;
  hideSettings?: boolean;
};
const Header: FunctionComponent<Props> = ({
  children,
  hideSettings = false
}) => {
  const { data, loading } = useQuery<GetMe>(GetMeQuery);
  const isAtVersions = useRouteMatch(ROUTE.VERSIONS)?.isExact;

  if (loading)
    return <div className={styles.splash} data-testid="splashscreen" />;

  const username = get(data?.me, 'email', '');

  return (
    <header className={styles.container} data-testid="app-header">
      <Link to={ROUTE.HOME} className={cx({[styles.linkDisabled]: isAtVersions})}>
        <img
          className={styles.konstellationText}
          src={'/img/brand/konstellation.svg'}
          alt="konstellation"
        />
      </Link>
      <div className={styles.customHeaderElements}>{children}</div>
      {!hideSettings && <Settings label={username} />}
    </header>
  );
};

export default memo(Header);
