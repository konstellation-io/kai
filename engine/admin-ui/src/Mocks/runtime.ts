import { runtime, version } from './version';

import { loader } from 'graphql.macro';

const GetRuntimeAndVersionQuery = loader(
  'Graphql/queries/getRuntimeAndVersions.graphql'
);

export const runtimeAndVersions = {
  runtime,
  versions: [version, version]
};
export const getRuntimeAndVersionsMock = {
  request: {
    query: GetRuntimeAndVersionQuery
  },
  result: {
    data: runtimeAndVersions
  }
};

export const getRuntimeAndVersionsErrorMock = {
  request: {
    query: GetRuntimeAndVersionQuery
  },
  error: new Error('cannot get runtime and versions')
};
