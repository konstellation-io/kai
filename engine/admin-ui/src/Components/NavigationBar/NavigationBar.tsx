import { ErrorMessage, SpinnerCircular, VerticalBar } from 'kwc';
import React, { memo } from 'react';

import ConditionalLink from '../ConditionalLink/ConditionalLink';
import { GetVersionConfStatus } from 'Graphql/queries/types/GetVersionConfStatus';
import HexButton from './HexButton';
import { NavLink } from 'react-router-dom';
import ROUTE from 'Constants/routes';
import { buildRoute } from 'Utils/routes';
import { loader } from 'graphql.macro';
import styles from './NavigationBar.module.scss';
import { useQuery } from '@apollo/client';

const GetRuntimeQuery = loader('Graphql/queries/getRuntimeAndVersions.graphql');

function NavigationBar() {
  const { data, loading, error } = useQuery<GetVersionConfStatus>(
    GetRuntimeQuery
  );

  if (loading || !data) return <SpinnerCircular />;
  if (error) return <ErrorMessage />;

  return (
    <VerticalBar>
      <div className={styles.container} data-testid="navigation-bar">
        <ConditionalLink
          LinkType={NavLink}
          to={buildRoute.runtime(ROUTE.RUNTIME, data.runtime.id)}
          linkProps={{
            activeClassName: styles.active,
            className: styles.link
          }}
        >
          <HexButton label={data.runtime.name} />
        </ConditionalLink>
      </div>
    </VerticalBar>
  );
}

export default memo(NavigationBar);
