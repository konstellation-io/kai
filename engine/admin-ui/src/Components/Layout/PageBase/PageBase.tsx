import React, { ReactElement, useEffect, useState } from 'react';
import Header from '../../Header/Header';
import cx from 'classnames';
import styles from './PageBase.module.scss';
import { useReactiveVar } from '@apollo/client';
import { logTabs } from '../../../Graphql/client/cache';

type PageBaseProps = {
  children: ReactElement | ReactElement[] | null;
  headerChildren?: ReactElement | ReactElement[] | null;
  customClassname?: string;
};

function PageBase({
  children,
  headerChildren,
  customClassname
}: PageBaseProps) {
  const [opened, setOpened] = useState<boolean>(false);

  const dataLogTabs = useReactiveVar(logTabs);

  useEffect(() => {
    setOpened(!!dataLogTabs.length);
  }, [dataLogTabs]);

  return (
    <>
      <Header>{headerChildren}</Header>
      <div
        className={cx(styles.content, customClassname, {
          [styles.withLogs]: opened
        })}
      >
        {children}
      </div>
    </>
  );
}

export default PageBase;
