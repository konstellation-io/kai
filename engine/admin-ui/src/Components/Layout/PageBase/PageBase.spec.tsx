import { BrowserRouter } from 'react-router-dom';
import Header from '../../Header/Header';
import { MockedProvider } from '@apollo/client/testing';
import NavigationBar from '../../NavigationBar/NavigationBar';
import PageBase from './PageBase';
import React from 'react';
import { mountApolloComponent } from 'Utils/testUtilsEnzyme';
import { usernameMock } from 'Mocks/auth';

const mocks = [usernameMock];

function Wrapper({ mocks }: any) {
  return (
    <BrowserRouter>
      <MockedProvider mocks={mocks} addTypename={false}>
        <PageBase
          headerChildren={<span className="headerChildren">Child</span>}
        >
          <div className="element">Element</div>
        </PageBase>
      </MockedProvider>
    </BrowserRouter>
  );
}
const Component = <Wrapper mocks={mocks} />;

describe('PageBase', () => {
  let wrapper: any;

  beforeEach(async () => {
    wrapper = await mountApolloComponent(Component);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains right components', () => {
    expect(wrapper.exists(Header)).toBeTruthy();
    expect(wrapper.exists(NavigationBar)).toBeTruthy();
    expect(wrapper.exists('.element')).toBeTruthy();
  });

  it('allows the addition of children to the header', () => {
    expect(wrapper.find(Header).exists('.headerChildren')).toBeTruthy();
    expect(wrapper.find('.content').exists('.headerChildren')).toBeFalsy();
  });
});
