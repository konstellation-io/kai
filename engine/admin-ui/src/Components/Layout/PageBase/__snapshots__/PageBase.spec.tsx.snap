// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBase matches snapshot 1`] = `undefined`;
